src/services/service.py

```diff
@@ -11,7 +23,72 @@ class FeedService(Service):
 
 
 class NewsService(Service):
-    pass
+
+    def __init__(self):
+        self._feed_repository = FeedRepository()
+        self._news_repository = NewsRepository()
+        self.upper_publish_date_boundary = datetime.now().replace(hour=0, minute=0, second=0, microsecond=0)
+        self.lower_publish_date_boundary = self.upper_publish_date_boundary - timedelta(days=1)
+
+    def scrap_and_save_news(self):
+        news_list = self.scrap_news_from_all_feeds()
+        return self._news_repository.save_all(news_list)
+
+    def scrap_news_from_all_feeds(self):
+        feed_list = self._feed_repository.get_all()
+
+        news_list = []
+
+        for feed in feed_list:
+            news_from_feed = self.scrap_news_from_feed(feed)
+            news_list = news_list + news_from_feed
```

Comment:
Could be replaced with list comprehension:
`news_list = [*self.scrap_news_from_feed(feed) for feed in feed_list]` but i don't know if we can sacrifice readability a bit

Reply:
As discussed we do not change this fragment of code.